res/controllers/Traktor-Kontrol-S2-MK2-hid-scripts.js

```diff
@@ -0,0 +1,1336 @@
+        sGroup, "play", makeSamplerCallback(group, i)));
+      TraktorS2MK2.pad_connections[group].push(engine.makeConnection(
+        sGroup, "repeat", makeSamplerCallback(group, i)));
+    }
+  }
+  
+  TraktorS2MK2.pad_connections[group].forEach(function(connection) {
+    connection.trigger();
+  });
+  
+  TraktorS2MK2.current_pad_mode[group] = padMode;
+}
+
+TraktorS2MK2.pad = function(field) {
+  var group = field.id.split(".")[0];
+  var buttonNumber = parseInt(field.name[field.name.length - 1]);
+  var padMode = TraktorS2MK2.current_pad_mode[group];
+  
+  if (padMode === TraktorS2MK2.pad_modes.hotcue) {
+    if (TraktorS2MK2.shift_pressed[group]) {
+      engine.setValue(field.group, "hotcue_" + buttonNumber + "_clear", field.value);
+    } else {
+      engine.setValue(field.group, "hotcue_" + buttonNumber + "_activate", field.value);
+    }
+  } else if (padMode === TraktorS2MK2.pad_modes.intro_outro) {
+    if (TraktorS2MK2.shift_pressed[group]) {
+      engine.setValue(field.group, intro_outro_keys[buttonNumber-1] + "_clear", field.value);
+    } else {
+      engine.setValue(field.group, intro_outro_keys[buttonNumber-1] + "_activate", field.value);
+    }
+  } else if (padMode === TraktorS2MK2.pad_modes.sampler) {
+    if (field.value === 0) {
+      return;
+    }
+    var samplerNumber = group === "[Channel1]" ? buttonNumber : buttonNumber + 4;
+    var samplerGroup = "[Sampler" + samplerNumber + "]";
+    if (TraktorS2MK2.shift_pressed[group]) {
+      if (engine.getValue(samplerGroup, "play") === 1) {
+        engine.setValue(samplerGroup, "play", 0);
+      } else {
+        script.triggerControl(samplerGroup, "eject");
+      }
+    } else {
+      if (engine.getValue(samplerGroup, "track_loaded") === 0) {
+        script.triggerControl(samplerGroup, "LoadSelectedTrack");
+      } else {
+        script.triggerControl(samplerGroup, "cue_gotoandplay");
+      }
+    }
+  }
+}
+
+TraktorS2MK2.samplerModeButton = function(field) {
+  if (field.value === 0) {
+    return;
+  }
+  var padMode = TraktorS2MK2.current_pad_mode[field.group];
+  if (padMode !== TraktorS2MK2.pad_modes.sampler) {
+    TraktorS2MK2.setPadMode(field.group, TraktorS2MK2.pad_modes.sampler);
+    TraktorS2MK2.controller.setOutput(field.group, "!remix_button", 0x7f, false);
+    TraktorS2MK2.controller.setOutput(field.group, "!flux_button", 0x00, !TraktorS2MK2.freeze_lights);
+  } else {
+    TraktorS2MK2.setPadMode(field.group, TraktorS2MK2.pad_modes.hotcue);
+    TraktorS2MK2.controller.setOutput(field.group, "!remix_button", 0x00, !TraktorS2MK2.freeze_lights);
+  }
+}
+
+TraktorS2MK2.introOutroModeButton = function(field) {
+  if (field.value === 0) {
+    return;
+  }
+  var padMode = TraktorS2MK2.current_pad_mode[field.group];
+  if (padMode !== TraktorS2MK2.pad_modes.intro_outro) {
+    TraktorS2MK2.setPadMode(field.group, TraktorS2MK2.pad_modes.intro_outro);
+    TraktorS2MK2.controller.setOutput(field.group, "!flux_button", 0x7f, false);
```

Comment:
what is the flux button supposed to do? I have one on my S3 and I don't get it.  Also too, what's a "grid" button?

Reply:
This is what flux is supposed to do: https://bugs.launchpad.net/mixxx/+bug/1475303